File: com.issue.tracker.api.persistence/src/main/java/com/issue/tracker/api/persistence/auth/AuthDsGateway.java
package com.issue.tracker.api.persistence.auth;

import jakarta.ejb.Remote;

@Remote
public interface AuthDsGateway {
    boolean existsByUsernameAndPassword(String username, String password);

    UserDsCompleteResponseModel findByUsername(String username);

    UserDsCompleteResponseModel save(SaveUserRequestModel saveUserRequestModel);

    UserDsCompleteResponseModel findById(Long id);

    UserDsCompleteResponseModel findByEmail(String email);
}

File: com.issue.tracker.api.persistence/src/main/kotlin/com/issue/tracker/api/persistence/auth/AuthDsGateway.kt
package com.issue.tracker.api.persistence.auth

interface AuthDsGateway {
    fun existsByUsernameAndPassword(username: String, password: String): Boolean

    fun findByUsername(username: String): UserDsResponseModel?

    fun save(saveUserRequestModel: SaveUserRequestModel): UserDsResponseModel?

    fun findById(id: Long): UserDsResponseModel?
}